main.cpp

```diff
@@ -2717,6 +2717,28 @@ uint32_t guess_flash_size(memory_access &access) {
     return size * 2;
 }
 
+// returns true if filename is a hex string, and fills array with the values
+bool filename_to_hex_array(uint8_t idx, uint8_t *array, size_t size) {
+    auto filename = settings.filenames[idx];
```

Comment:
IMHO this function would feel more self-contained if it was directly passed the string, instead of being passed an index into an array? :confused: